### PR TITLE
make _pushNotification() smarter

### PR DESCRIPTION
**Pull request by AitonGoldman**
_pushNotification() will push a notification onto the stack, and set a timeout to remove the notification from the stack.  When it's time to remove the notification, if it doesn't find the notification in the stack it will still remove the first item in the stack.  This fixes that.

### CODE REVIEW

**Comment by caroso1222** [comment]
Hey @AitonGoldman thanks for catching it! I can't see anything wrong so I'm merging as is but, just wondering, do you have any use case where the notification stack will not find the notification? Just wanted to be sure I'm not missing any other problem you may have seen.

Thanks for contributing!

File: src/notyf.ts [hunk 1]
```diff
@@ -48,7 +48,9 @@ export default class Notyf {
     const duration = notification.options.duration || this.options.duration;
     setTimeout(() => {
       const index = this.notifications.indexOf(notification);
-      this.notifications.splice(index, 1);
+      if(index!=1){
```
Comment (caroso1222) [inline src/notyf.ts]: I think you meant to evaluate against `-1`, right?
```suggestion
      if(index !== -1){
```

**Comment by caroso1222** [comment]
I just pointed out something very minor which might have been a typo btw. `index === 1` is valid as notifications can have different durations. Here's an example of a valid `=== 1` index. I think you meant to compare against `-1` to avoid removing the first element in the stack, as suggested.

**Comment by AitonGoldman** [comment]
My application will remove notifications from the stack when certain events happen - and those events can happen before the timeout that is set by notyf on notification creation.

**Comment by caroso1222** [comment]
@AitonGoldman gotcha, that makes sense. Alright then, could you please confirm that the suggestion I made on the PR is valid? I suppose you wanted to compare to `-1` which is when `indexOf` can't find the value.

**Pull request merged**